netinfo/__init__.py

```diff
@@ -56,14 +57,18 @@ def get_ifnames():
     return ifnames
 
 
-class Error(Exception):
-    pass
+def get_hostname() -> str:
+    return socket.gethostname()
+
+
+def get_fqdn() -> str:
+    return socket.getfqdn()
 
 
-class InterfaceInfo(object):
+class InterfaceInfo:
     """enumerate network related configurations"""
 
-    sockfd = lazyclass(socket.socket)(socket.AF_INET, socket.SOCK_DGRAM)
+    sockfd = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
```

Comment:
This changes behaviour. Lazy class only creates the socket when it's first accessed where as this creates it when the class is created, even if it's never used. If we're not using lazyclass and not re-creating it's lazy-initialization functionality, it would be nest to create it as an instance variable rather than a class variable.

That said there's issues with the original code too because it never closes the socket. Afaik the only "proper" way to handle this is to do it via a context manager using the `__enter__` and `__exit__` dunder methods, although for our purposes using a `__del__` should work fine.

Reply:
Ah crap. I tested it and it appeared to work fine, but you raise a really good point that I hadn't considered!

Context manager FTW! :+1: 

Unfortunately, I'll need to put this on the backburner for now as it's not really urgent and there are much higher priority issues to deal with...

Reply:
Resolved by implementing lazyclass